src/Boson_Sampling_Utilities.py

```diff
@@ -121,3 +122,108 @@ def generate_lossy_inputs(initial_state: ndarray, number_of_particles_left: int)
             lossy_inputs_list.append(lossy_input)
 
     return lossy_inputs_list
+
+
+def calculate_number_of_possible_n_particle_m_mode_output_states(n: int, m: int) -> int:
+    """
+        Calculates the number of possible output states with n particles placed around m modes. This is basically
+        the same answer as to in how many possible combinations can we put n objects in m bins. It's also a dimension
+        of n-particle m-mode bosonic space. Stars-and-bars argument applies here.
+    """
+    return binom(n + m - 1, n)
+
+
+class ChinHuhPermanentCalculator:
+
+    def __init__(self, matrix: ndarray, input_state: List[int] = [], output_state: List[int] = []):
+        self.__matrix = matrix
+        self.__input_state = input_state
+        self.__output_state = output_state
+
+    @property
+    def matrix(self) -> ndarray:
+        return self.__matrix
+
+    @matrix.setter
+    def matrix(self, matrix) -> None:
+        self.__matrix = matrix
+
+    @property
+    def input_state(self) -> List[int]:
+        return self.__input_state
+
+    @input_state.setter
```

Comment:
Are you really using the setter? Ain't all of the operations performed just getter on the list and performing an operation on it?

Reply:
Maybe I misunderstood something, but I believe that I use setters, but in a non-explicit way. What I mean there's no C++ syntax like
`object.setProperty(...)`
but rather
`object.property = ...`.
Should I not use setters and getters in this case?

Reply:
You should use setters in case you're using `object.property = ...`. But in this case, having a list as a property, do you actually use the assignment?

Reply:
Yea. I do it in permanent calculation tests (for example :P), but I imagine I will use it more as the project grows.